### PR TITLE
Zierman fix remove friend

### PR DESCRIPTION
**Pull request by Zierman**
With these changes all of the tests (in automated_tests.py) now pass.

### CODE REVIEW

File: database/db.py [hunk 1]
```diff
@@ -492,7 +492,8 @@ def remove_friend(self, username: str, friends_username: str) -> None:
         friends_id = friends_id[0][0]
         user_id = user_id[0][0]
         was_successful = self.db_delete(self.query_builder.remove_friend(user_id, friends_id))
-        self.db_delete(self.query_builder.remove_friend(friends_id, user_id))
+        if was_successful:
+            was_successful = all([was_successful, bool(self.db_delete(self.query_builder.remove_friend(friends_id, user_id)))])
```
Comment (Zierman) [inline database/db.py:497]: This makes sure that we only report success if both delete operations are successful.

File: user/friends_management.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def remove_friend(self, req_item: RemoveFriendRequest):
         if self.validate_username(friends_username):
             try:
                 self.db.remove_friend(username, friends_username)
+                req_item.set_response()
```
Comment (Zierman) [inline user/friends_management.py:97]: This was the reason the test failed (in addition to not having delete privileges earlier). Previously the response was not set when the operation was successful. I must have missed it when I was reworking the message item stuff.

File: data/message_item.py [hunk 3]
```diff
@@ -537,8 +537,6 @@ def set_response(self, *,
         if not failure_reason:
             if token:
                 if data and len(data) > 0 and len(data[0]) > 7:
-                    logger.debug(data)
```
Comment (Zierman) [inline data/message_item.py:540]: I removed this log as it is unlabeled data.

File: data/responder.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def send_account_creation_status(self, connection_socket: socket, status):
         connection_socket.send(status.encode())
 
     def send_response(self, msg_item: BaseRequest, timeout_seconds: float):
-        logger.debug(msg_item.response)
+        logger.debug(f'response: {msg_item.response}')
```
Comment (Zierman) [inline data/responder.py:36]: I updated this log message so we know what it is (response).

File: database/db.py [hunk 5]
```diff
@@ -333,7 +333,7 @@ def check_if_friend_request_exists(self, sender_username: str, recipient_usernam
         user_id = user_id[0][0]
         friends_id = friends_id[0][0]
         result = self.db_fetch(self.query_builder.check_if_friend_request_exists(user_id, friends_id))
-        logger.debug(type(result))
+        logger.debug(f'type of result from check_if_friend_request_exists: {type(result)}')
```
Comment (Zierman) [inline database/db.py:336]: I updated this log message so we know what it is.

File: database/mysql_db.py [hunk 6]
```diff
@@ -226,7 +226,6 @@ def db_connection(self) -> MySQLConnection:
     def db_connection(self, connection: MySQLConnection):
         self._db_connection = connection
         self._cursor = self._db_connection.cursor()
-        logger.debug(f'cursor set to {self._cursor} in db_connect setter')
```
Comment (Zierman) [inline database/mysql_db.py:229]: I removed this debug log message, I don't need it any more

File: database/mysql_db.py [hunk 7]
```diff
@@ -332,19 +331,15 @@ def user(self) -> str:
     def context(self) -> MySQLContext:
         return self._context
 
-    @logged_method
```
Comment (Zierman) [inline database/mysql_db.py:335]: I don't need this logged anymore

File: database/mysql_db.py [hunk 8]
```diff
@@ -332,19 +331,15 @@ def user(self) -> str:
...
     def __enter__(self) -> MySQLContext:
         self._context = MySQLContext()
         self._context.db_connection = mysql.connector.connect(user=self.user, password=self.password,
                                                               host=self.host,
                                                               database=self.database_name,
                                                               auth_plugin=self.auth_plugin)
-        log(f'db_connection is {self._context.db_connection}')
-        log(f'cursor is {self._context.cursor}')
```
Comment (Zierman) [inline database/mysql_db.py:343]: I don't need these logged anymore

File: database/mysql_db.py [hunk 9]
```diff
@@ -332,19 +331,15 @@ def user(self) -> str:
...
         return self._context
 
     # noinspection PyBroadException
-    @logged_method
```
Comment (Zierman) [inline database/mysql_db.py:347]: I don't need this logged anymore

File: process_request.py [hunk 10]
```diff
@@ -63,11 +63,9 @@ def process_requests(self):
 
         while True:
             request_item = self.request_queue.get()
-            logger.debug('a')  # TODO Remove
```
Comment (Zierman) [inline process_request.py:66]: removed old debug log message.

File: process_request.py [hunk 11]
```diff
@@ -63,11 +63,9 @@ def process_requests(self):
...
             # Decrypt parsed_data
             try:
                 self.process_request_type(request_item)
-                logger.debug('b')  # TODO Remove
```
Comment (Zierman) [inline process_request.py:70]: removed old debug log message

File: user/signin.py [hunk 12]
```diff
@@ -46,7 +46,7 @@ def signin(self, req_item: SigninRequest):
                 # Bundle the token into the response package
                 signon_token = self.db.get_token(username)
                 signon_token = signon_token[0][0]
-                logger.debug(type(signon_token))
+                logger.debug(f'type of token from signin: {type(signon_token)}')
```
Comment (Zierman) [inline user/signin.py:49]: made it so we can tell what this is logging.

**Review by atormanen — APPROVED** [review]

**Pull request merged**